homeassistant/components/iqvia/__init__.py

```diff
@@ -116,53 +104,58 @@ def __init__(self, client, sensor_types):
 
     async def async_update(self):
         """Update IQVIA data."""
-        from pyiqvia.errors import InvalidZipError
+        condition_method_mapping = {
+            (TYPE_ALLERGY_FORECAST,): [
+                (self._client.allergens.extended, TYPE_ALLERGY_FORECAST),
```

Comment:
We could store the string `'allergens.extended'`, instead of the actual attribute, and split the string on period then use `getattr` on each item in the split string. That way we could break out the dictionary to a constant at the module level.

Is it too convoluted?

Reply:
A more readable way could be to use our handler registry helper.

https://github.com/home-assistant/home-assistant/blob/7a6acca6bb1f2aca2f7f977c20cb513f5f331328/homeassistant/util/decorator.py#L7-L17

- Register a handler function for each allergy type string.
- Make the mapping dict just between the sensor types and the allergy types.
- Iterate the dict checking the condition and look up the correct handler if the condition passes (or inverted as guard clause).
- Collect the selected handler tasks.

Reply:
I think I see what you're after. Let me give it a shot and have you tell me whether I've understood you correctly.

Reply:
Check out 88f2a25 – thoughts?